server/src/main/webapp/WEB-INF/rails/webpack/views/pages/clicky_pipeline_config/modal/material_modal.tsx

```diff
@@ -71,9 +72,18 @@ export class MaterialModal extends Modal {
   }
 
   body(): m.Children {
+    const allScmMaterialsHaveDestination = this.materials().scmMaterialsHaveDestination();
+    const maybeMsg                       = this.isNew && !allScmMaterialsHaveDestination
+      ? <FlashMessage type={MessageType.warning} dataTestId={"materials-destination-warning-message"}>
+        In order to configure multiple SCM materials for this pipeline, each of its material needs have to a 'Destination Directory' specified. Please
+        edit the existing material and specify a 'Destination Directory' in order to proceed with this operation.
+      </FlashMessage>
+      : undefined;
```

Comment:
should we use `if` for better readability?

Reply:
Is the multi-line message an issue or the involvement of two boolean variables?

Reply:
Umm, in this particular scenario, 
the condition has two parts, `this.isNew && !allScmMaterialsHaveDestination` which itself was complex to understand.
Also, the ternary contains `undefined` as one of the value evaluation, Where in javascript, if a variable isn't initialized, by default it has `undefined` value. Which makes the `undefined` part of the ternary redundant.